forge-gui/res/cardsfolder/upcoming/sarinth_steelseeker.txt

```diff
@@ -0,0 +1,12 @@
+Name:Sarinth Steelseeker
+ManaCost:1 G
+Types:Creature Human Artificer Scout
+PT:1/2
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Artifact.YouCtrl | TriggerZones$ Battlefield | Execute$ TrigDig | TriggerDescription$ Whenever an artifact enters the battlefield under your control, look at the top card of your library. If it's a land card, you may reveal it and put it into your hand. If you don't put the card into your hand, you may put it into your graveyard.
+SVar:TrigDig:DB$ PeekAndReveal | PeekAmount$ 1 | RevealValid$ Land | RevealOptional$ True | RememberRevealed$ True | SubAbility$ DBChangeZone
```

Comment:
We need to somehow remember it whether it was revealed or not or line 8 won't always work

probably ImprintRevealed + RememberPeeked